### PR TITLE
perf: Load SAPUI5 types only when needed

### PR DESCRIPTION
**Pull request by matz3**

### CODE REVIEW

File: src/detectors/typeChecker/host.ts [hunk 1]
```diff
@@ -5,6 +5,84 @@ import fs from "node:fs/promises";
 import {createRequire} from "node:module";
 const require = createRequire(import.meta.url);
 
+const SAPUI5_TYPES_LIBS = [
```
[the remark this answers is not in the document]
Comment (matz3) [reply src/detectors/typeChecker/host.ts:8]: or better: read fs for available `d.ts` files in `@sapui5/types/types/` folder

[the remark this answers is not in the document]
Comment (matz3) [reply src/detectors/typeChecker/host.ts]: done

**Comment by d3xter666** [comment]
LGTM

I've done a perf measurement with a lint against `openui5/sap.m`. Here are the results

| Command | Mean [s] | Min [s] | Max [s] | Relative |
|:---|---:|---:|---:|---:|
| `ui5lint` (`lazy-load-sapui5-dts`) | 33.412 ± 0.392 | 33.118 | 34.425 | 1.00 |
| `ui5lint` (`main`) | 33.752 s ±  0.466 | 32.983 | 34.645 | 1.00 |

**Comment by matz3** [comment]
> Command	Mean [s]	Min [s]	Max [s]	Relative
> `ui5lint` (`lazy-load-sapui5-dts`)	33.412 ± 0.392	33.118	34.425	1.00
> `ui5lint` (`main`)	33.752 s ± 0.466	32.983	34.645	1.00

Thanks for doing a measurement. I just did some manual tests and for me the benefit was about 5s for sap.m.
I wonder why you didn't get such a difference.

File: src/detectors/typeChecker/host.ts [hunk 2]
```diff
@@ -31,6 +31,28 @@ async function collectTransitiveDependencies(pkgName: string, deps: Set<string>)
 	return deps;
 }
 
+async function collectSapui5TypesFiles() {
+	const typesDir = path.dirname(require.resolve("@sapui5/types/package.json"));
+	const typesFiles = await fs.readdir(path.join(typesDir, "types"), {withFileTypes: true});
+	return typesFiles
+		.filter((entry) => entry.isFile() && entry.name.endsWith(".d.ts") && entry.name !== "index.d.ts")
+		.map((entry) => entry.name);
```
[the remark this answers is not in the document]
Comment (matz3) [reply src/detectors/typeChecker/host.ts]: We'll probably never have more than 100 entries here and the function is only called once per linting, but I guess it makes sense to be consistent and prevent multiple looping over arrays in general.

**Review by RandomByte — APPROVED** [review]
LGTM

**Pull request merged**